source/common/config/utility.h

```diff
@@ -531,6 +537,45 @@ class Utility {
     }
     return std::make_unique<FixedBackOffStrategy>(dns_refresh_rate_ms);
   }
+
+  /**
+   * Prepares backoff strategy from envoy::config::core::v3::ApiConfigSource
+   * @param config envoy::config::core::v3::ApiConfigSource
+   * @param random random generator
+   * @return BackOffStrategyPtr
+   */
+  static BackOffStrategyPtr
+  prepareBackoffStrategy(const envoy::config::core::v3::ApiConfigSource& api_config_source,
```

Comment:
The input is very specific to config-source, and it should probably be more generalized. Seems similar to the code [here](https://github.com/envoyproxy/envoy/blob/5991a2053e981038c105a421886af073be9c5941/source/common/config/datasource.cc#L70-L80).

Reply:
done, refactored to a generalized implementation and reused in `source/common/config/datasource.cc`